### PR TITLE
firebase-1.4-code

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const admin = require('firebase-admin');
+const { firebaseDb } = require("../config/firebaseConfig");
 const signUp = async (req, res) => {
-  const { email, password } = req.body;
+  const { email, password, address, age, gender, name } = req.body;
```
Comment (amrit56789) [inline controllers/userController.js]: Fields should be firstName, lastName, MobileNo, email, password, confirmPassword, profilePhoto.

File: controllers/userController.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const admin = require('firebase-admin');
+const { firebaseDb } = require("../config/firebaseConfig");
 const signUp = async (req, res) => {
```
Comment (amrit56789) [inline controllers/userController.js:3]: userSignUp

File: controllers/userController.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
-  const { email, password } = req.body;
+  const { email, password, address, age, gender, name } = req.body;
   if (!email || !password) {
```
Comment (amrit56789) [inline controllers/userController.js:6]: Put a password condition means check password and confirmPassword.

File: controllers/userController.js [hunk 4]
```diff
@@ -9,7 +10,14 @@ const signUp = async (req, res) => {
       email,
       password,
     });
-    res.status(200).send(user)
+    const userDocRef = firebaseDb.collection("users").doc(user.uid);
```
Comment (amrit56789) [inline controllers/userController.js]: not firebaseDB It is a firestore.

File: controllers/userController.js [hunk 5]
```diff
@@ -9,7 +10,14 @@ const signUp = async (req, res) => {
...
+    await userDocRef.set({
```
Comment (amrit56789) [inline controllers/userController.js]: Write a function in one line.

File: index.js [hunk 6]
```diff
@@ -2,6 +2,7 @@ const express = require("express")
 const bodyParser = require("body-parser")
 require("dotenv").config();
 const userRoutes = require("./routes/userRoutes")
+require("./config/firebaseConfig")
```
Comment (amrit56789) [inline index.js]: require?

File: controllers/userController.js [hunk 7]
```diff
@@ -1,15 +1,31 @@
...
       email,
       password,
     });
-    res.status(200).send(user)
+    fireStore.collection("users").doc(user.uid).set({
+      firstName,
+      address,
+      lastName,
+    });
+    const uploadProfilePhoto = async (file, filename) => {
```
Comment (yuvrajInno) [inline controllers/userController.js]: move this code to a saparete file, so any function can use this @ashok-choudhary-0

**Review by yuvrajInno — CHANGES_REQUESTED** [review]

**Pull request merged**